src/DataAccess/DoctrineDonationRepository.php

```diff
@@ -136,4 +140,117 @@ private function getDataFieldsFromAddress( PhysicalAddress $address ) {
 		];
 	}
 
+	/**
+	 * @param int $id
+	 *
+	 * @return Donation|null
+	 * @throws GetDonationException
+	 */
+	public function getDonationById( int $id ) {
+		try {
+			/**
+			 * @var DoctrineDonation $donation
+			 */
+			$donation = $this->entityManager->find( DoctrineDonation::class, $id );
+		}
+		catch ( ORMException $ex ) {
+			throw new GetDonationException( $ex );
+		}
+
+		if ( $donation === null ) {
+			return null;
+		}
+
+		return $this->newDonationDomainObject( $donation );
+	}
+
+	private function newDonationDomainObject( DoctrineDonation $dd ): Donation {
+		$donation = new Donation();
+
+		$donation->setId( $dd->getId() );
+		$donation->setStatus( $dd->getStatus() );
+		$donation->setAmount( $dd->getAmount() );
+		$donation->setInterval( $dd->getPeriod() );
+		$donation->setPaymentType( $dd->getPaymentType() );
+		$donation->setBankTransferCode( $dd->getTransferCode() );
+		$donation->setOptsIntoNewsletter( $dd->getInfo() );
+
+		$donation->setPersonalInfo( $this->getPersonalInfoFromEntity( $dd ) );
+		$donation->setBankData( $this->getBankDataFromEntity( $dd ) );
+		$donation->setTrackingInfo( $this->getTrackingInfoFromEntity( $dd ) );
+
+		return $donation;
+	}
+
+	private function getPersonalInfoFromEntity( DoctrineDonation $dd ): PersonalInfo {
+		$personalInfo = new PersonalInfo();
+
+		$personalInfo->setEmailAddress( $dd->getEmail() );
+		$personalInfo->setPersonName( $this->getPersonNameFromEntity( $dd ) );
+		$personalInfo->setPhysicalAddress( $this->getPhysicalAddressFromEntity( $dd ) );
+		$personalInfo->setPhysicalAddress( $this->getPhysicalAddressFromEntity( $dd ) );
+		$personalInfo->setPhysicalAddress( $this->getPhysicalAddressFromEntity( $dd ) );
```

Comment:
You could also extract it into a method. Something like `hitThePersonalInfoWithLeadPipeUntilItFinallySetsTheAddress()`.